internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,356 @@
+		}
+		qs = append(qs, newClusterNameQuestion(clusterName, message))
+	}
+
+	if opts.Provider == "" {
+		qs = append(qs, newClusterProviderQuestion())
+	}
+
+	if err := survey.Ask(qs, opts); err != nil {
+		return err
+	}
+
+	if regionQ := newRegionQuestions(opts.Region, opts.Provider); regionQ != nil {
+		// we call survey.Ask two times because the region question needs opts.Provider to be populated
+		if err := survey.Ask([]*survey.Question{regionQ}, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// askDBUserAccessListFlags allows the user to set required flags by using interactive prompts
+func (opts *Opts) askDBUserAccessListFlags() error {
+	var qs []*survey.Question
+
+	message := "Insert the Username for authenticating to MongoDB"
+	dbUser := opts.DBUsername
+	if dbUser == "" {
+		dbUser = dbUsername()
+		if dbUser != "" {
+			message = fmt.Sprintf("Insert the Username for authenticating to MongoDB [Press Enter to use '%s']", dbUser)
+		}
+
+		qs = append(qs, newDBUsernameQuestion(dbUser, message, opts.validateUniqueUsername))
+	}
+
+	if opts.DBUserPassword == "" {
+		qs = append(qs, newDBUserPasswordQuestion())
+	}
+
+	if len(opts.IPAddresses) == 0 {
+		message = "Insert the IP entry to add to the Access List"
+		publicIP := net.IPAddress()
+		if publicIP != "" {
+			message = fmt.Sprintf("Insert the IP entry to add to the Access List [Press Enter to use your public IP address '%s']", publicIP)
+		}
+		q := newAccessListQuestion(publicIP, message)
+		qs = append(qs, q)
+	}
+
+	if len(qs) > 0 {
+		if err := survey.Ask(qs, opts); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.IPAddresses) == 0 && opts.IPAddress != "" {
+		opts.IPAddresses = []string{opts.IPAddress}
+	}
+
+	if opts.DBUserPassword == "" {
+		// The user wants to auto-generate the password
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) validateUniqueUsername(val interface{}) error {
+	username, _ := val.(string)
+	dbUser, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), username)
+	if err != nil {
+		if !strings.Contains(err.Error(), fmt.Sprintf("No user with username %s exists.", username)) {
+			return err
+		}
+	}
+
+	if dbUser != nil {
+		return errors.New("a user with this username already exists")
+	}
+
+	return nil
+}
+
+// dbUsername returns the username of the user by running the command 'whoami'
+func dbUsername() string {
+	userStruct, err := user.Current()
+	if err != nil {
+		return ""
+	}
+
+	// dbUsername can only contain ASCII letters, numbers, hyphens and underscores
+	out := strings.TrimSpace(userStruct.Username)
+	var re = regexp.MustCompile("([^A-Za-z0-9_-])")
+	return re.ReplaceAllString(out, "_")
+}
+
+// newClusterName returns an auto-generate Cluster name
+func (opts *Opts) newClusterName() string {
+	cs, _ := opts.store.ProjectClusters(opts.ConfigProjectID(), nil)
```

Comment:
[nit] you should check this error, what happens on a network error for example?

Reply:
See: https://github.com/mongodb/mongocli/pull/512#discussion_r590393087. We wouldn’t go inside the for loop and the function would return `""`

Reply:
my question stand, let's say there's a network error, you don't get the list, it probably is null, or empty, what happens? this method should return an error as well as an empty string if somethings goes wrong you may not be able to safely assume a name

Reply:
in case of an error this piece of code `if clusters, ok := cs.([]atlas.Cluster); ok {` would be false so the function would return an empty string. Therefore the UI would show the message `"Insert the cluster name"` instead of `"Insert the cluster name [Press Enter to use the auto-generated name '%s']"`

```
	clusterName := opts.ClusterName
	if clusterName == "" {
		clusterName = opts.newClusterName()
		if clusterName != "" {
			message = fmt.Sprintf("Insert the cluster name [Press Enter to use the auto-generated name '%s']", clusterName)
		}
		qs = append(qs, newClusterNameQuestion(clusterName, message))
	}
```